spec/jobs/upload_set_update_job_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'spec_helper'
+
+describe UploadSetUpdateJob do
+  let(:user) { FactoryGirl.find_or_create(:jill) }
+  let(:upload_set) { UploadSet.create }
+
+  let!(:file) do
+    GenericFile.new(upload_set: upload_set) do |file|
```

Comment:
You can use `#create` instead of `#new` here (and then you can drop the `#save!`